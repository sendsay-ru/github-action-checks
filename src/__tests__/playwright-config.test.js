const { render } = require('../utils');
const source = require('../../config/source');
const playwrightConfig = require('../../config/playwright');

describe('Render by playwright config', () => {
  it('should expose a function', () => {
    expect(render).toBeDefined();
  });

  it('should return expected value', async () => {
    expect(await render(source, playwrightConfig)).toMatchSnapshot();
  });
});
