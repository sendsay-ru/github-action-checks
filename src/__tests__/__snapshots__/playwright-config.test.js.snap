// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render by playwright config should return expected value 1`] = `
[
  {
    "content": "# rendered by https://www.npmjs.com/package/github-actions-checks
name: Checks
on:
  pull_request:
  push:
    branches:
      - stable
      - master
      - main
    tags:
      - v**

concurrency:
  group: \${{ github.workflow }}-\${{ github.event.pull_request.number }}
  cancel-in-progress: true

jobs:
  e2e:
    runs-on: ubuntu-latest

    outputs:
      playwright: \${{ steps.playwright.outputs.result }}

    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Preparing
        uses: ./.github/actions/check-install

      - name: Pre command
        if: always()
        shell: bash
        run: npx playwright install --with-deps

      - name: Playwright run
        id: playwright
        if: always()
        uses: ./.github/actions/check-run
        with:
          script: npm run playwright test

      - uses: actions/upload-artifact@v3
        if: always()
        with:
          name: artifacts
          retention-days: 30
          path: |
            playwright-report
            playwright-artifacts
",
    "output": ".github/workflows/checks.yml",
  },
  {
    "content": "# rendered by https://www.npmjs.com/package/github-actions-checks
name: Preparing
description: preparing to run the checks

runs:
  using: composite
  steps:
    - name: Setup node
      uses: actions/setup-node@v3
      with:
        node-version: 16

    - name: Install packages
      shell: bash
      run: npm install
",
    "output": ".github/actions/check-install/action.yml",
  },
  {
    "content": "# rendered by https://www.npmjs.com/package/github-actions-checks
name: Run check
description: run check script

inputs:
  script:
    required: true
    description: script command

outputs:
  result:
    description: "Script result"
    value: \${{ steps.result.outputs.status }}

runs:
  using: composite
  steps:
    - name: Run script
      id: script
      shell: bash
      run: \${{ inputs.script }}

    - name: Results
      id: result
      if: always()
      env:
        SUCCESS_ICON: "☘️"
        FAILURE_ICON: "🐙"
      shell: bash
      run: |
        if [[ \${{ steps.script.outcome }} == 'success' ]]; then echo "status=$SUCCESS_ICON" >> $GITHUB_OUTPUT; else echo "status=$FAILURE_ICON" >> $GITHUB_OUTPUT; fi
",
    "output": ".github/actions/check-run/action.yml",
  },
]
`;
